forte/data/data_store.py

```diff
@@ -309,7 +310,10 @@ def add_annotation_raw(self, type_id: int, begin: int, end: int) -> int:
         # annotation type entry data with default fields.
         # A reference to the entry should be store in both self.__elements and
         # self.__entry_dict.
-        raise NotImplementedError
+        entry = self._new_annotation(type_id, begin, end)
```

Comment:
Now I feel there is a problem, where we store the `type_id` into the entry itself since we convert it to an integer.

What if we update the ontology with new types, will the `type_id` keep consistent? 

@wanglec

Reply:
I think we could handle updated ontology in `EntryTypeGenerator`. If we update ontology, we check the current dictionary which maps `type_id` and `entry_type`. If it exists in the dictionary, we do nothing. If it does not exist, we add it to the dictionary with a new `type_id`. Does this seem to work for you?

Reply:
not sure yet. that actually means that in the generator we somehow need to ensure all the`type_id` are the same for the same time from last time (say some existing data pack serialization on disk). This sounds hard to manage unless we have a deterministic entry name to id generator

Reply:
This is mainly related to the serialization/deserialization part. Maybe you can think about the workflow while deserializing a pack. There needs to be a `dict` after deserialization that maps from "entry type -> integer ids", and after that you deserialize this data pack, you need to make sure to align this mapping with the current mapping in memory. (i.e align the mapping from disk with the mapping from memory).

Reply:
Yes, do you mean to serialize this map somehow and use it when deserializing the data store?

Reply:
Yes, we will need to consider this. We need to have a mechanism to ensure the ids mapped to the correct type during steps like serialization.